contracts/vaults/RibbonThetaVault.sol

```diff
@@ -318,18 +325,66 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
 
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
-                round: _round,
+                round: currentRound,
                 amount: uint128(newAmount)
             });
         } else {
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
-                round: _round,
+                round: currentRound,
                 amount: uint128(amount)
             });
         }
 
-        totalPending = totalPending.add(amount);
+        _totalPending = _totalPending.add(amount);
+
+        // If we have an unprocessed pending deposit from the previous rounds, we have to redeem it.
+        if (depositReceipt.round < currentRound && !depositReceipt.processed) {
+            _redeemDeposit(currentRound, depositReceipt);
+        }
+    }
+
+    function redeemDeposit() external nonReentrant {
+        VaultDeposit.DepositReceipt memory depositReceipt =
+            depositReceipts[msg.sender];
+        _redeemDeposit(round, depositReceipt);
+    }
+
+    function _redeemDeposit(
+        uint16 currentRound,
+        VaultDeposit.DepositReceipt memory depositReceipt
+    ) private {
+        require(!depositReceipt.processed, "Processed");
+        require(depositReceipt.round > currentRound, "Round not closed");
+
+        uint256 pps = roundPricePerShare[currentRound];
+        // If this throws, it means that vault's roundPricePerShare[currentRound] has not been set yet
+        // which should never happen.
+        // Has to be larger than 1 because `1` is used in `initRoundPricePerShares` to prevent cold writes.
+        require(pps > 1, "Invalid pps");
+
+        depositReceipts[msg.sender].processed = true;
+
+        uint256 shares = wmul(depositReceipt.amount, pps);
+
+        emit Redeem(msg.sender, shares, depositReceipt.round);
+
+        transfer(msg.sender, shares);
+    }
+
+    function withdrawInstantly(uint256 amount) external nonReentrant {
+        VaultDeposit.DepositReceipt storage depositReceipt =
+            depositReceipts[msg.sender];
+
+        require(!depositReceipt.processed, "Processed");
+        require(depositReceipt.round == round, "Invalid round");
+
+        // Subtraction underflow checks already ensure it is smaller than uint128
+        depositReceipt.amount = uint128(
+            uint256(depositReceipt.amount).sub(amount)
+        );
+
```

Comment:
would it be cheaper to do depositReceipts[msg.sender].processed, etc everywhere instead of creating storage variable pointer depositReceipt?

Reply:
Do you mean `depositReceipts[msg.sender].amount`? There isn't a difference because it's just a single `SSTORE` opcode